orttraining/orttraining/test/training_api/common/synthetic_data_loader.h

```diff
@@ -48,24 +41,14 @@ struct SyntheticInput {
     return shape_;
   }
 
- protected:
-  std::vector<int64_t> shape_;
-  size_t num_of_elements_{1};
-};
-
-template <typename T>
-struct TypedSyntheticInput : public SyntheticInput {
-  explicit TypedSyntheticInput(const std::vector<int64_t>& shape)
-      : SyntheticInput(shape) {
-    data_.resize(num_of_elements_);
-  }
-
-  std::vector<T>& Data() {
+  SyntheticDataVector& GetData() {
     return data_;
   }
 
  private:
-  std::vector<T> data_;
+  std::vector<int64_t> shape_;
```

Comment:
TensorShapeVector

Reply:
note that std::vector and TensorShapeVector own their storage, whereas gsl::span is just a view over memory. so if changing to gsl::span, need to ensure that the memory stays valid. given we don't want to depend on internal ORT types, std::vector seems fine here in my opinion.

Reply:
I realized that after I pushed the fix, will fix it later. Thanks man! :)